dds/DCPS/WriteDataContainer.h

```diff
@@ -528,6 +531,28 @@ class OpenDDS_Dcps_Export WriteDataContainer : public PoolAllocationBase {
   DDS::DurabilityServiceQosPolicy const & durability_service_;
 
 #endif
+
+  /// Timer responsible for reporting missed offered deadlines.
+  RcHandle<DCPS::PmfSporadicTask<WriteDataContainer> > deadline_task_;
+  TimeDuration deadline_period_; // TimeDuration::zero_value means no deadline.
+  typedef std::multimap<MonotonicTimePoint, PublicationInstance_rch> DeadlineMapType;
+  DeadlineMapType deadline_map_;
+
+  /// Lock for synchronization of @c status_ member.
+  ACE_Recursive_Thread_Mutex& deadline_status_lock_;
```

Comment:
Yuck. Understandable, but yuck.

Reply:
Yeah.  Passing around references to locks...what could possibly go wrong?